src/panels/config/automation/condition/ha-automation-condition.ts

```diff
@@ -122,12 +123,56 @@ export default class HaAutomationCondition extends LitElement {
     }
   }
 
-  public expandAll() {
-    const rows = this.shadowRoot!.querySelectorAll<HaAutomationConditionRow>(
+  private _getRowElements() {
+    return this.shadowRoot!.querySelectorAll<HaAutomationConditionRow>(
       "ha-automation-condition-row"
-    )!;
-    rows.forEach((row) => {
+    );
+  }
+
+  private _getChildCollapsableElements(
+    row: HaAutomationConditionRow
+  ): NodeListOf<HaAutomationCondition> | undefined {
+    const editor = row.shadowRoot!.querySelector<HaAutomationConditionEditor>(
+      "ha-automation-condition-editor"
+    );
+    if (editor) {
+      const buildingBlock = editor.shadowRoot?.querySelector(
+        CONDITION_BUILDING_BLOCKS.map(
+          (block) => `ha-automation-condition-${block}`
+        ).join(", ")
+      );
+      if (buildingBlock) {
+        return buildingBlock.shadowRoot?.querySelectorAll<HaAutomationCondition>(
+          "ha-automation-condition"
+        );
```

Comment:
I don't like this, why not call a `expandAll` function on `ha-automation-condition-row` that then calls an `expandAll` function on `ha-automation-condition-editor`, etc. That way each component is responsible for it's own children and `ha-automation-condition` doesnt have to know how all the children of children, of children are handled. If something is changed in a child this will cause problems.